fix `gh pr edit`: do not fetch V1 projects on unsupported GitHub hosts

BagToad opened the pull request:
This pull request updates how project v1 support is handled when editing issues and pull requests. The main change is to pass explicit project v1 support information through the editing workflow, ensuring that project v1 options are only fetched and presented when supported by the repository. This affects function signatures, calls, and related tests for both issues and pull requests.

Fixes #11986

BagToad committed:
Pass ProjectsV1Support to FetchOptions functions

Update FetchOptions and related function signatures to accept a ProjectsV1Support parameter, enabling conditional logic based on project support. This change improves flexibility for handling project fields in issue and PR editing flows.

BagToad committed:
Inject Detector mock in edit command tests

Added Detector field with EnabledDetectorMock to EditOptions in issue and PR edit command tests to ensure feature detection logic is exercised during test runs. Also updated EditableOptionsFetch signature in pr/edit/edit_test.go to include projectsV1Support parameter for compatibility with shared.FetchOptions.

BagToad commented on pkg/cmd/pr/shared/editable.go:
```diff
@@ -413,7 +414,7 @@ func FetchOptions(client *api.Client, repo ghrepo.Interface, editable *Editable)
 		Assignees:      editable.Assignees.Edited,
 		ActorAssignees: editable.Assignees.ActorAssignees,
 		Labels:         editable.Labels.Edited,
-		ProjectsV1:     editable.Projects.Edited,
+		ProjectsV1:     editable.Projects.Edited && projectV1Support == gh.ProjectsV1Supported,
```
This is the actual meat and potatoes. The rest of the changes are just to satisfy the interface change.

babakks reviewed (approved):
LGTM! Can we also have a test to verify the Projects V1 data are not fetched?

BagToad committed:
Add test for non-interactive projects v1 unsupported

Introduces a test case to ensure that when editing pull requests non-interactively and projects v1 is unsupported, v1 project metadata is not fetched and only v2 project queries are made.

jbmichaels1984 reviewed (commented):
Android 16 install free

cortezsrrobert2-dev reviewed (commented):
#11987 

- [ ] ### [](url)

netabs909122-spec reviewed (commented):
pl help me

netabs909122-spec commented:
pl help me

Treetosayoot commented:
> This pull request updates how project v1 support is handled when editing issues and pull requests. The main change is to pass explicit project v1 support information through the editing workflow, ensuring that project v1 options are only fetched and presented when supported by the repository. This affects function signatures, calls, and related tests for both issues and pull requests.
> 
> Fixes #11986

Treetosayoot reviewed (commented):
Born to be free

DrHeadhunter reviewed (commented):
Ich
